astropy/utils/iers/tests/test_iers.py

```diff
@@ -149,9 +149,16 @@ def test_simple(self):
         assert status3 == iers.FROM_IERS_A_PREDICTION
         assert ut1_utc3 != 0.
 
+
 class TestIERS_Auto():
 
     @remote_data
```

Comment:
For whatever reason, the test only fails without the patched code if IERS_A is downloaded (thus your need for "remote_data"). This also passes interactively in IPython using the unpatched code:
```python
with iers.conf.set_temp('iers_auto_url', None):
    with iers.conf.set_temp('auto_download', False):
        t = iers.IERS_Auto.open()
assert type(t) is iers.IERS_B
```

Maybe this is a symptom of a deeper bug?